components/popover/popover-mixin.js

```diff
@@ -794,33 +831,55 @@ export const PopoverMixin = superclass => class extends superclass {
 
 		const pointerRect = pointer.getBoundingClientRect();
 
-		if (this._preferredPosition.span !== 'all') {
-			const xAdjustment = Math.min(20 + ((pointerRotatedLength - pointerLength) / 2), (openerRect.width - pointerLength) / 2);
-			if (!this._rtl) {
-				if (this._preferredPosition.span === 'end') {
-					position.left = openerRect.left + xAdjustment;
+		if (this._location === 'block-end' || this._location === 'block-start') {
```

Comment:
The only difference here is that we're accounting for the scrollbar width when RTL.